server/methods/core/orders.js

```diff
@@ -310,50 +321,44 @@ Meteor.methods({
   "orders/sendNotification": function (order) {
     check(order, Object);
 
-    // just make sure this a real userId
-    // todo: ddp limit
-    if (!Meteor.userId()) {
-      throw new Meteor.Error(403, "Access Denied");
+    if (!this.userId) {
+      Logger.error("orders/sendNotification: Access denied");
+      throw new Meteor.Error("access-denied", "Access Denied");
     }
 
     this.unblock();
-    if (order) {
-      let shop = Shops.findOne(order.shopId);
-      let shipment = order.shipping[0];
-
-      Reaction.configureMailUrl();
-      Logger.info("orders/sendNotification", order.workflow.status);
-      // handle missing root shop email
-      if (!shop.emails[0].address) {
-        shop.emails[0].address = "no-reply@reactioncommerce.com";
-        Logger.warn("No shop email configured. Using no-reply to send mail");
-      }
-      // anonymous users without emails.
-      if (!order.email) {
-        Logger.warn("No shop email configured. Using anonymous order.");
-        return true;
-      }
-      // email templates can be customized in Templates collection
-      // loads defaults from reaction-email-templates/templates
-      let tpl = `orders/${order.workflow.status}`;
-      SSR.compileTemplate(tpl, ReactionEmailTemplate(tpl));
-      try {
-        return Email.send({
-          to: order.email,
-          from: `${shop.name} <${shop.emails[0].address}>`,
-          subject: `Order update from ${shop.name}`,
-          html: SSR.render(tpl, {
-            homepage: Meteor.absoluteUrl(),
-            shop: shop,
-            order: order,
-            shipment: shipment
-          })
-        });
-      } catch (error) {
-        Logger.fatal("Unable to send notification email: " + error);
-        throw new Meteor.Error("error-sending-email", "Unable to send order notification email.", error);
-      }
+
+    const shop = Shops.findOne(order.shopId);
+    const shipment = order.shipping[0];
+
+    Logger.info(`orders/sendNotification status: ${order.workflow.status}`);
+
+    // handle missing root shop email
+    if (!shop.emails[0].address) {
+      shop.emails[0].address = "no-reply@reactioncommerce.com";
+      Logger.warn("No shop email configured. Using no-reply to send mail");
+    }
+
+    // anonymous users without emails.
+    if (!order.email) {
+      const msg = "No order email found. No notification sent.";
+      Logger.warn(msg);
+      throw new Meteor.Error("email-error", msg);
     }
+
+    // email templates can be customized in Templates collection
+    // loads defaults from /private/email/templates
+    const tpl = `orders/${order.workflow.status}`;
+    SSR.compileTemplate(tpl, Reaction.Email.getTemplate(tpl));
+
+    Reaction.Email.send({
```

Comment:
Another place we will need translations, perhaps another method like "Reaction.email.submit" that had all the translation logic before Reaction.email.send could make this a bit easier to replace through the codebase, and give a convenient location for future email transforms.